### DISCUSSION TITLE
Crop images in 2 folders based on trimmed transparent crop of image A

### DISCUSSION THREAD

**Discussion by justifun** [ImageMagick/ImageMagick#7463] [Help]
### ImageMagick version

7.1.1.1-34-q16-hdri

### Operating system, version and so on

windows 11

### Description

I have 2 folders A and B full of png images with transparent backgrounds

I'd like to crop image A using the mogrify -trim command to remove all of the transparent pixels, AND then apply the same crop coordinates to matching image B in the other folder so that the result would be 2 images both cropped the same width and height so that they both line up if i were to layer then down the road....

eg: open file 1 in folder A (crop to image 1 bounding box) - store the cropped coordinates and apply those values to image 1 in folder B

thx

**Comment by fmw42** [comment]
Don't use magick mogrify.  Use simply magick and save the trim results to a variable to use for the second image.  Assuming the two images are the same size

magick imageA.suffix -format "%@\n" -write info: -trim +repage outputA.suffix

That should both trim and get the trim values. The %@ gets the trim values without trimming. Then the -trim does the trimming You should be able to save the trim values to a variable.  Sorry, I am not a Windows user and so do not know how to save to variable.

Alternately, you could do the following:

`magick imageA.suffix -set option:trimvals "%@" -crop "%[trimvals]" +repage -write outputA.suffix ( imageB.suffix -crop "%[trimvals]" +repage -write outputB.suffix ) null:`

This reads imageA and stores the trimvals, then crops imageA using the trimvals and saves the output. Then in the parens, it reads imageB and crops using the same trimvals and save the output. Finally it writes a null: image to exit.

**Reply by justifun** [reply]
Awesome thanks!  Small bug though, when it gets to the second pair of images eg: A2, B2, it still uses the cropping from A1.

**Reply by fmw42** [reply]
> AND then apply the same crop coordinates to matching image B

I thought that was what you asked.  So what exactly do you want?

**Reply by justifun** [reply]
Sorry if I wasn't clear enough.  In each of the 2 source folders (A and B) the first Image A should crop to its bounding box, and the first image of folder B should use its cropping.  This is working with the code you provided.

There are dozens of files in each of the A and B folders though.
So on the second file. it should do the same thing.  A2 crops to its bounding box and B2 uses that cropping area. and so on

So i just need it to loop through all of them and each file pair goes together.  A1 + B1, A2 + B2 etc.

thank you for your time.

**Comment by snibgo** [comment]
In fmw42's second suggestion, I think there is a typo, and the second `outputA.suffix` should be `outputB.suffix`. So the suggestion would be:
```
magick imageA.suffix -set option:trimvals "%@" -crop "%[trimvals]" +repage -write outputA.suffix ( imageB.suffix -crop "%[trimvals]" +repage -write outputB.suffix ) null:
```
If this is used in a Windows BAT file, then each `%` should be doubled, like this:
```
magick imageA.suffix -set option:trimvals "%%@" -crop "%%[trimvals]" +repage -write outputA.suffix ( imageB.suffix -crop "%%[trimvals]" +repage -write outputB.suffix ) null:
```
Each `imageA.suffix` etc should be a filename, possibly with drive and directory.

That command will only process one pair of files. To process all the pairs, the command could be used inside a `for` loop. I suppose all the corresponding pairs of files in the two input directories have the same names. Do you know how to use `for`?

**Reply by fmw42** [reply]
Yes, sorry for the type. In my second command the second output should be outputB, as snibgo points out.  I have edited my answer to correct that.
